### PR TITLE
#19452 Not use dotRaw when the field is a integer

### PR DESCRIPTION
**Pull request by freddyucv**
Not use dotRaw when the field is a intege

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java [hunk 1]
```diff
@@ -26,6 +27,7 @@
     private boolean unique = Boolean.FALSE;
     private String contentTypeId;
     private String relationType;
+    private DataTypes dataType;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:30]: Codacy found an issue: [Field dataType has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java [hunk 2]
```diff
@@ -1,20 +1,16 @@
 package com.dotmarketing.cms.urlmap;
 
 import static com.dotcms.datagen.TestDataUtils.getNewsLikeContentType;
-import static org.jgroups.util.Util.assertEquals;
-import static org.jgroups.util.Util.assertFalse;
+import static org.jgroups.util.Util.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:4]: Codacy found an issue: [Avoid unused imports such as 'org.jgroups.util.Util'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java [hunk 3]
```diff
@@ -1,20 +1,16 @@
...
+import com.dotcms.contenttype.model.field.DataTypes;
+import com.dotcms.contenttype.model.field.Field;
 import com.dotcms.contenttype.model.type.ContentType;
-import com.dotcms.datagen.FolderDataGen;
-import com.dotcms.datagen.HTMLPageDataGen;
-import com.dotcms.datagen.SiteDataGen;
-import com.dotcms.datagen.TemplateDataGen;
-import com.dotcms.datagen.TestDataUtils;
-import com.dotcms.datagen.UserDataGen;
+import com.dotcms.datagen.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java [hunk 4]
```diff
@@ -1,12 +1,13 @@
 package com.dotmarketing.cms.urlmap;
 
+import com.dotcms.content.elasticsearch.constants.ESMappingConstants;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.constants.ESMappingConstants'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java [hunk 5]
```diff
@@ -668,4 +659,43 @@ private UrlMapContext getUrlMapContext(final User systemUser, final Host host, f
...
     }
+
+    /**
+     * methodToTest {@link URLMapAPIImpl#processURLMap(UrlMapContext)}
+     * Given Scenario: Process a URL Map url when both the Content Type and Content exists but the field to Map is a
+     * {@link com.dotcms.contenttype.model.field.DataTypes#INTEGER}
+     * ExpectedResult: Should return a {@link URLMapInfo} wit the right content ans detail page
+     */
+    @Test
+    public void shouldReturnContentletWhenTheContentExistsAndUseAIntegerFIeld()
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:670]: Please add a test for a DataTypes.FLOAT as well

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java [hunk 6]
```diff
@@ -205,17 +206,30 @@ private String buildFields(final ContentType structure,
...
-            query.append('+').append(structure.variable()).append('.')
-                    .append(fieldMatches.get(counter)).append("_dotRaw").append(':')
-                    .append(ESUtils.escapeExcludingSlashIncludingSpace(value)).append(' ');
+            query.append('+')
+                    .append(structure.variable()).append('.');
+
+            final String variableName = fieldMatches.get(counter);
+            final Field field = fieldMap.get(variableName);
+
+            if (field.dataType().equals(DataTypes.INTEGER)){
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java:225]: would it work for DataType.FLOAT

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 5 above)
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:670]: Is there any test for the happy path (non-number)

**Review by nollymar — CHANGES_REQUESTED** [review]
Please take a look at the comments I left

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 5 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:670]: Yes

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java [hunk 7]
```diff
@@ -668,4 +659,81 @@ private UrlMapContext getUrlMapContext(final User systemUser, final Host host, f
...
+        final URLMapInfo urlMapInfo = urlMapInfoOptional.get();
+        assertEquals(newsTestContent.getLongProperty(field.variable()),
+                urlMapInfo.getContentlet().getLongProperty(field.variable()));
+        assertEquals("/news-events/news/news-detail", urlMapInfo.getIdentifier().getURI());
+    }
+
+    /**
+     * methodToTest {@link URLMapAPIImpl#processURLMap(UrlMapContext)}
+     * Given Scenario: Process a URL Map url when both the Content Type and Content exists but the field to Map is a
+     * {@link com.dotcms.contenttype.model.field.DataTypes#INTEGER}
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:705]: Update this comment

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 7 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java]: done https://github.com/dotCMS/core/pull/19459/commits/f17437f7adc0402fb60bbb10ce4d32760dd8a60d#diff-de0fb9193432881efb85f6e38c30df1524a2a3e392f54d744734dd37520586daR705

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.constants.ESMappingConstants'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:30]: Codacy found an issue: [Field dataType has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:4]: Codacy found an issue: [Avoid unused imports such as 'org.jgroups.util.Util'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:30]: Codacy found an issue: [Field dataType has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:4]: Codacy found an issue: [Avoid unused imports such as 'org.jgroups.util.Util'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.constants.ESMappingConstants'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:4]: Codacy found an issue: [Avoid unused imports such as 'org.jgroups.util.Util'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.constants.ESMappingConstants'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:30]: Codacy found an issue: [Field dataType has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:4]: Codacy found an issue: [Avoid unused imports such as 'org.jgroups.util.Util'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.constants.ESMappingConstants'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java [hunk 8]
```diff
@@ -24,6 +25,8 @@
...
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:28]: Codacy found an issue: [Field dataType has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6354090)

**Comment by yolabingo** [comment]
cloud support - see ticket https://dotcms.zendesk.com/agent/tickets/102349

**Pull request merged**